proposals/NNNN-unlock-existentials-with-fixed-associated-types.md

```diff
@@ -0,0 +1,253 @@
+# Unlock Existentials with Fixed Associated Types
+
+* Proposal: [SE-NNNN](NNNN-unlock-existentials-with-fixed-associated-types.md)
+* Authors: [Filip Sakel](https://github.com/filip-sakel), [Anthony Latsis](https://github.com/AnthonyLatsis)
+* Review Manager: TBD
+* Status: **Awaiting implementation**
+
+## Introduction
+
+Today, Swift protocols are divided into two categories: those that _can_ be used as fully-fledged types, and those that can only be used as generic constraints because they have `Self` or associated type requirements. However, in some cases where a protocol inherits and fixes the associated types of another protocol - that doesn't have `Self` requirements, this constraint seems rather unintuitive. This proposal aims to relax this needless contraint allowing such protocols to be used as Types.
+
+## Motivation
+
+Currently, any protocol that has associated type requirements is not allowed to be used as a Type. That's a sensible constraint; however, considering that the associated types of some protocols are known, this restriction can become quite frustrating: 
+
+```swift
+protocol Identifiable {
+    associatedtype ID: Hashable
+    
+    var id: ID { get }
+}
+
+protocol User: Identifiable 
+    where ID == String {
+    
+    var username: String { get }
+}
+
+extension User {
+    var id: String {
+        username
+    }
+}
+
+let myUser: User
+// ❌ `User` is not
+// usable as a Type.
+```
+
+Many would rightfully assume that `User` could be used as a Type, since the only associated type (`ID`) is known to be `String` via a same-type constraint on the `User` protocol.
```

Comment:
Do you have a reason to uppercase words like Protocol, Type or Existential?

Reply:
It's just to draw emphasis on them, but they're probably better lowercased.